spec/lib/jobs/install-os-job-spec.js

```diff
@@ -87,21 +93,138 @@ describe('Install OS Job', function () {
+            expect(subscribeRequestPropertiesStub).to.have.been.called;
+            expect(subscribeHttpResponseStub).to.have.been.called;
+
+            cb = subscribeRequestProfileStub.firstCall.args[0];
+            expect(cb).to.be.a.function;
+            expect(cb.call(job)).to.equal(job.profile);
+
+            cb = subscribeRequestPropertiesStub.firstCall.args[0];
+            expect(cb).to.be.a.function;
+            expect(cb.call(job)).to.equal(job.options);
+
+            cb = subscribeHttpResponseStub.firstCall.args[0];
+            expect(cb).to.be.a.function;
+        });
+    });
 
-        cb = subscribeHttpResponseStub.firstCall.args[0];
-        expect(cb).to.be.a.function;
+    describe('test _convertInstallDisk', function() {
+        var catalog = {
+            data: [
+                {
+                    identifier: 0,
+                    linuxWwid: '/dev/test0',
+                    esxiWwid: 't10.abcde'
+                },
+                {
+                    identifier: 1,
+                    linuxWwid: '/dev/test1',
+                    esxiWwid: 'naa.rstuvw'
+                },
+                {
+                    identifier: 2,
+                    linuxWwid: '/dev/test2',
+                    esxiWwid: 'naa.xyzopq'
+                }
+            ]
+        };
+
+        beforeEach(function() {
+            job = new InstallOsJob(
+                {
+                    profile: 'testprofile',
+                    completionUri: '',
+                    version: '7.0',
+                    repo: 'http://127.0.0.1:8080/myrepo/7.0/x86_64',
+                    rootPassword: 'rackhd',
+                    rootSshKey: null,
+                    users: [
+                        {
+                            name: 'test',
+                            password: 'testPassword',
+                            uid: 100,
+                            sshKey: ''
+                        }
+                    ],
+                    dnsServers: null,
+                    installDisk: 1,
+                    osType: 'esx'
+                },
+                {
+                    target: 'testid'
+                },
+                uuid.v4());
+            waterline.catalogs.findMostRecent = sinon.stub().resolves(catalog);
+        });
+
+        afterEach(function() {
+            waterline.catalogs.findMostRecent.reset();
+        });
+
+        it('should set correct installDisk esxi wwid', function() {
+            return Promise.resolve().then(function() {
+                return job._convertInstallDisk();
+            }).then(function() {
+                expect(job.options.installDisk).to.equal('naa.rstuvw');
+            });
+        });
+
+        it('should set correct installDisk linux wwid', function() {
+            job.options.osType = 'linux';
+            return Promise.resolve().then(function() {
+                return job._convertInstallDisk();
```

Comment:
FYI, a nice way you can also do this is like:

`return expect(job._convertInstallDisk()).to.become('/dev/test1');`